.github/workflows/main.yml

```diff
@@ -3,19 +3,35 @@ name: Docker image CI
 on:
   push:
     branches: ['master']
-  workflow_dispatch:
+  
+env:
+  IMAGE_TAG: latest
+  REPO_URL: ${{ github.repository }}
+    
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+  
+      - name: Checkout code 
+        uses: actions/checkout@v3
+    
+      - name: Extract branch name
+        id: extract_branch
+        run: echo "BRANCH_NAME=$(basename ${{ github.ref }})" >> $GITHUB_ENV
+      
+      - name: build Docker image
+        run: |
+          docker build . --tag ghcr.io/${{ env.REPO_URL }}:${{ env.IMAGE_TAG }}-${BRANCH_NAME}
+      
```

Comment:
Tag will like `latest-master` or `latest-2020-01.01`

Reply:
Interesting so it auto builds the tags too. That seems convenient. I think the name is fine with the latest in it.